rocky/reports/report_types/systems_report/report.py

```diff
@@ -0,0 +1,67 @@
+from dataclasses import dataclass
+from datetime import datetime
+from logging import getLogger
+from typing import Any, Dict
+
+from django.utils.translation import gettext_lazy as _
+
+from octopoes.models import Reference
+from octopoes.models.ooi.dns.zone import Hostname
+from octopoes.models.ooi.network import IPAddressV4, IPAddressV6
+from reports.report_types.definitions import Report
+
+logger = getLogger(__name__)
+
+
+@dataclass
+class System:
+    system_types: list
+    oois: list
+
+
+class SystemReport(Report):
+    id = "systems-report"
+    name = _("System Report")
+    description = _("Combine ip addresses, hostnames and services into systems.")
+    plugins = {"required": [], "optional": []}
+    input_ooi_types = {Hostname, IPAddressV4, IPAddressV6}
+    template_path = "systems_report/report.html"
+
+    def generate_data(self, input_ooi: str, valid_time: datetime) -> Dict[str, Any]:
+        reference = Reference.from_str(input_ooi)
+        ips = []
+
+        if reference.class_type == Hostname:
+            ips = self.octopoes_api_connector.query(
+                "Hostname.<hostname[is ResolvedHostname].address", valid_time, reference
+            )
+        elif reference.class_type in [IPAddressV4, IPAddressV6]:
```

Comment:
Small detail and doesn't apply to only this specific example; but generally if you have constants or data that are not meant to be changed in the first place, then prefer tuples over lists

```suggestion
        elif reference.class_type in (IPAddressV4, IPAddressV6):
```